.github/workflows/release.yaml

```diff
@@ -0,0 +1,84 @@
+name: Release
+
+on:
+    push:
+        branches: [ main, master ]
+    pull_request:
+    release:
+        types: [ created ]
+
+jobs:
+    build-phar:
+        runs-on: ubuntu-latest
+        name: Build PHAR
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v3
+                with:
+                    fetch-depth: 0
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: '8.1'
+                    ini-values: phar.readonly=0
+                    tools: composer
+                    coverage: none
+
+            -   name: Install Composer dependencies
+                uses: ramsey/composer-install@v2
+
+            -   name: Build PHAR
+                run: |
+                    make phpbrew
+                    mv phpbrew bin/phpbrew.phar
+
+            # Smoke test
+            -   name: Ensure the PHAR works
+                run: bin/phpbrew.phar --version
+
+            -   name: Import GPG key
+                if: github.event_name == 'release'
+                uses: crazy-max/ghaction-import-gpg@v5
+                with:
+                    gpg_private_key: ${{ secrets.GPG_KEY_<TODO:privatekeyID> }}
+                    passphrase: ${{ secrets.GPG_KEY_<TODO:privatekeyID>_PASSPHRASE }}
+
+            -   name: Sign the PHAR
+                if: github.event_name == 'release'
+                run: |
+                    gpg --local-user <TODO:email> \
+                        --batch \
+                        --yes \
+                        --passphrase="${{ secrets.GPG_KEY_<TODO:privatekeyID>_PASSPHRASE }}" \
+                        --detach-sign \
+                        --output bin/phpbrew.phar.asc \
```

Comment:
To be consistency, I think it should be the `bin/phpbrew.asc`.

Reply:
Note that tools like PHIVE enforces the convention `path/to/your.phar` & `path/to/your.phar.asc`; but that can be visited separately